### PR TITLE
Add product to shopping cart feature

### PR DESCRIPTION
**Pull request by janekkerner**
Add product to shopping cart feature, sprint 4, task 3

### CODE REVIEW

**Review by kamilpuchala — COMMENTED** [review]
Your code looks good!
I left two comments, please check them, especially  from AddProductToCart service

File: app/services/shopping_cart_services/add_product_to_cart.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+module ShoppingCartServices
+  class AddProductToCart
+    def call(cart, product)
+      if cart.cart_items.exists?(product_id: product.id)
```
[the remark this answers is not in the document]
Comment (janekkerner) [reply app/services/shopping_cart_services/add_product_to_cart.rb]: I wasn't sure about acceptance criteria, as there was only info about price and name in shopping cart. During implementation that was my biggest dilemma, as I thought that product quantity feature will be in next sprint. :) I need to ask on slack more often.
I did a few changes in that service and hope that it still doesn't look that bad.

File: spec/requests/shopping_cart_request_spec.rb [hunk 2]
```diff
@@ -23,4 +24,30 @@
       expect(response.body).to include('Your shopping cart is empty')
     end
   end
+
+  describe 'POST /cart' do
+    it 'has success http status when user is signed in' do
```
[the remark this answers is not in the document]
Comment (janekkerner) [reply spec/requests/shopping_cart_request_spec.rb]: He should be redirected to sign in page. I forgot about that option, as this happens many times through many features. I did tests update also on system spec.

**Review by kamilpuchala — APPROVED** [review]
Thx for changes GJ! 👍

**Pull request merged**